myhpom/tests/test_cloudfactory_tasks.py

```diff
@@ -0,0 +1,100 @@
+                document=SimpleUploadedFile(
+                    os.path.basename(PDF_FILENAME), open(PDF_FILENAME, 'rb').read()
+                ),
+                valid_date=now(),
+            )
+        )
+        self.task = CloudFactorySubmitDocumentRun
+
+    def test_201_created(self, reqmock):
+        response_data = json.load(
+            open(os.path.join(FIXTURE_PATH, 'cloudfactory', 'post_response_201.json'), 'rb')
+        )
+        reqmock.post(settings.CLOUDFACTORY_API_URL + '/runs', **response_data)
+        run_id = self.task(self.document_url.id)
+        self.assertTrue(reqmock.called)
+        self.assertEqual(type(run_id), int)  # self.task() returns the id of the run object
+        cf_run = self.document_url.cloudfactorydocumentrun_set.last()
+        self.assertIsNotNone(cf_run)
+        self.assertEqual(cf_run.status, response_data['json']['status'])
+        self.assertEqual(cf_run.run_id, response_data['json']['id'])
+        self.assertEqual(cf_run.created_at, parse_datetime(response_data['json']['created_at']))
+
+    def test_422_unprocessable(self, reqmock):
+        response_data_set = json.load(
+            open(os.path.join(FIXTURE_PATH, 'cloudfactory', 'post_responses_422.json'), 'rb')
+        )
+        for response_data in response_data_set.values():
+            reqmock.post(settings.CLOUDFACTORY_API_URL + '/runs', **response_data)
+            self.assertRaises(ValueError, self.task, self.document_url.id)
+            cf_run = self.document_url.cloudfactorydocumentrun_set.last()
+            self.assertIsNotNone(cf_run)
+            self.assertEqual(cf_run.status, CloudFactoryDocumentRun.STATUS_NEW)
+            self.assertIn("Invalid request.", cf_run.response_content)
+            self.assertIsNone(cf_run.run_id)
+            self.assertIsNone(cf_run.created_at)
+
+    def test_404_notfound(self, reqmock):
+        response_data = json.load(
+            open(os.path.join(FIXTURE_PATH, 'cloudfactory', 'post_response_404.json'), 'rb')
+        )
+        reqmock.post(settings.CLOUDFACTORY_API_URL + '/runs', **response_data)
+        self.assertRaises(ValueError, self.task, self.document_url.id)
+        cf_run = self.document_url.cloudfactorydocumentrun_set.last()
+        self.assertIsNotNone(cf_run)
+        self.assertEqual(cf_run.response_content, response_data['text'])
+
+    def test_non_json_response(self, reqmock):
+        response_data = json.load(
+            open(os.path.join(FIXTURE_PATH, 'cloudfactory', 'post_invalid_response_201.json'), 'rb')
+        )
+        reqmock.post(settings.CLOUDFACTORY_API_URL + '/runs', **response_data)
+        self.assertRaises(ValueError, self.task, self.document_url.id)
+
+    def test_connection_timeout(self, reqmock):
+        reqmock.post(
+            settings.CLOUDFACTORY_API_URL + '/runs', exc=requests.exceptions.ConnectTimeout
+        )
+        self.assertRaises(requests.exceptions.ConnectTimeout, self.task, self.document_url.id)
```

Comment:
Add assertions on the state of the run object.

Reply:
adding a STATUS_TIMEOUT to the run class and the task